common/formatting/token_partition_tree_test.cc

```diff
@@ -500,4 +500,762 @@ TEST_F(MergeConsecutiveSiblingsTest, TwoGenerations) {
+      tree_type{arg5},
+    },
+  };
+
+  verible::BasicFormatStyle style; // default
+  style.column_limit = 14 + 7;
+
+  tree.Children()[1].ApplyPreOrder(
+      [&style](TokenPartitionTree& node) {
+        auto& uwline = node.Value();
+        uwline.SetIndentationSpaces(7 + style.indentation_spaces);
+      });
+
+  tree.Children()[0].Value().SetIndentationSpaces(7);
+  tree.Value().SetIndentationSpaces(7);
+
+  UnwrappedLine group1(0, header.TokensRange().begin());
+  group1.SpanUpToToken(arg2.TokensRange().end());
+  UnwrappedLine group2(0, arg3.TokensRange().begin());
+  group2.SpanUpToToken(arg5.TokensRange().end());
+
+  tree_type tree_expected{
+    all,
+    tree_type{group1,
+      tree_type{header},
+      tree_type{arg1},
+      tree_type{arg2},
+    },
+    tree_type{group2,
+      tree_type{arg3},
+      tree_type{arg4},
+      tree_type{arg5},
+    },
+  };
+
+  auto saved_tree(tree);
+  ReshapeFittingSubpartitions(&tree, style);
+
+  const auto diff = DeepEqual(tree, tree_expected, TokenRangeEqual);
+  EXPECT_TRUE(diff.left == nullptr) << "Expected:\n" << tree_expected
+                                    << "\nGot:"      << tree << "\n";
+
+  // Check group nodes indentation
+  EXPECT_EQ(tree.Children()[0].Value().IndentationSpaces(), 7);
+  EXPECT_EQ(tree.Children()[1].Value().IndentationSpaces(),
+            header.TokensRange()[0].Length() + 7); // appended
+
+  EXPECT_EQ(saved_tree.Children()[0].Value().IndentationSpaces(),
+            tree.Children()[0].Children()[0].Value().IndentationSpaces());
+  EXPECT_EQ(saved_tree.Children()[1].Children()[0].Value().IndentationSpaces(),
+            tree.Children()[0].Children()[1].Value().IndentationSpaces());
+  EXPECT_EQ(saved_tree.Children()[1].Children()[1].Value().IndentationSpaces(),
+            tree.Children()[0].Children()[2].Value().IndentationSpaces());
+
+  EXPECT_EQ(saved_tree.Children()[1].Children()[2].Value().IndentationSpaces(),
+            tree.Children()[1].Children()[0].Value().IndentationSpaces());
+  EXPECT_EQ(saved_tree.Children()[1].Children()[3].Value().IndentationSpaces(),
+            tree.Children()[1].Children()[1].Value().IndentationSpaces());
+  EXPECT_EQ(saved_tree.Children()[1].Children()[4].Value().IndentationSpaces(),
+            tree.Children()[1].Children()[2].Value().IndentationSpaces());
+}
+
+// Tests with real-word example
+class ReshapeFittingSubpartitionsTestFixture : public ::testing::Test,
+                                               public UnwrappedLineMemoryHandler {
+ public:
+  ReshapeFittingSubpartitionsTestFixture()
+      // combining what would normally be a type and a variable name
+      // into a single string for testing convenience
+      : sample_("function_fffffffffff( "
+                "type_a_aaaa, type_b_bbbbb, "
+                "type_c_cccccc, type_d_dddddddd, "
+                "type_e_eeeeeeee, type_f_ffff);"),
+        tokens_(absl::StrSplit(sample_, ' ')) {
+    EXPECT_EQ(tokens_.size(), 7);
```

Comment:
`ASSERT_EQ` to stop test early if false

Reply:
Changing to `ASSERT_EQ` gives: `error: returning a value from a constructor`.
That `EXPECT_EQ` was really just for me and I think it's not really necessary.
I removed it.